### PR TITLE
Initial implementation of kubernetes-test snap

### PR DESCRIPTION
**Pull request by Cynerva**
Here's a first pass at creating a snap for e2e.

### CODE REVIEW

File: kubernetes-test.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: kubernetes-test
+version: 'KUBE_VERSION'
+summary: tests for kubernetes
+description: |
+  tests for kubernetes
+grade: devel
```
Comment (chuckbutler) [inline kubernetes-test.yaml]: Is this writing to a system space? I think we can get by with just the network plug and have it output results to `$SNAP_USER_DATA`.

File: ginkgo-e2e [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -eu
+
+GINKGO_ARGS="${GINKGO_ARGS:-}"
+
+# e2e.test needs to be placed in a directory we have write access to.
+# Otherwise, some of the tests will fail.
+TEMP_DIR="/tmp/cdk-test.ginkgo-e2e"
```
Comment (chuckbutler) [inline ginkgo-e2e]: Can this be reworked to leverage `$SNAP_DATA` or `$SNAP_USER_DATA` that way it retains strict confinement?

File: build [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -eux
+
+export KUBE_VERSION="${KUBE_VERSION:-$(curl -L https://dl.k8s.io/release/stable.txt)}"
```
Comment (chuckbutler) [inline build]: I like this fun little snippet :D :boom:

File: kubernetes-test.yaml (hunk 1 above)
Comment (Cynerva) [inline kubernetes-test.yaml]: I'm not aware of any reason strict confinement won't work - I just haven't tried it. I'll give it a shot.

I will need to pack the kubectl bin in the snap to make it work, for sure.

File: ginkgo-e2e (hunk 2 above)
Comment (Cynerva) [inline ginkgo-e2e]: I'm pretty sure strictly confined snaps get their own mount of /tmp. I.E. writing to `/tmp/` from the snap will actually write to `/tmp/snap.kubernetes-test.tmp/` on the host. Or something close to that, anyway. Does that resolve your concern?

I'd like to keep it in /tmp myself - seems like a more appropriate place for temporary data - but I'm open to moving it if you're still opposed.

Comment (chuckbutler) [inline ginkgo-e2e]: nope! if that works with strict confinement I'm +1 to that.

We don't -have- to retain strict confinement here, I think its fine to go with classic, but we need to fully understand why that's the case when we are striving for strict confinement across the board to help enable our security story.

**Comment by Cynerva** [comment]
Closing, this landed here: https://github.com/kubernetes/release/pull/293

**Pull request closed without merging**